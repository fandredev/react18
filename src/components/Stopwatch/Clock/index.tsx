import styles from './clock.module.scss'

interface ClockProps {
  time: number | undefined
}

export default function Clock({time = 0}: ClockProps) {
  const minutes = Math.floor(time / 60)
  const seconds = time % 60
  const [minTen, minUnity] = String(minutes).padStart(2,'0')
  const [secTen, secUnity] = String(seconds).padStart(2,'0')

  return (
    <>
      <span className={styles.clockNumber}>{minTen}</span>
      <span className={styles.clockNumber}>{minUnity}</span>
      <span className={styles.clockDivision}>:</span>
      <span className={styles.clockNumber}>{secTen}</span>
      <span className={styles.clockNumber}>{secUnity}</span>
    </>
  )
}

