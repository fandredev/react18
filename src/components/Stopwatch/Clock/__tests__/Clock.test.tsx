import { render } from "@testing-library/react";
import Clock from "../";


describe(`Tests from ${Clock.name}`, () => {
  it('Should be possible to snapshot the Clock component when the page is rendered', () => {
    const { container } = render(
      <Clock time={10} />
    )

    expect(container).toMatchSnapshot()
  })
})
