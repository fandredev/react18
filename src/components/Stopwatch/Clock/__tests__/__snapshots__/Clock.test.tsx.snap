// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests from Clock Should be possible to snapshot the Clock component when the page is rendered 1`] = `
<div>
  <span
    class="clockNumber"
  >
    0
  </span>
  <span
    class="clockNumber"
  >
    0
  </span>
  <span
    class="clockDivision"
  >
    :
  </span>
  <span
    class="clockNumber"
  >
    1
  </span>
  <span
    class="clockNumber"
  >
    0
  </span>
</div>
`;
